### ISSUE TITLE
No error thrown when function moved to new file

### ISSUE THREAD

**Issue by tanevanwifferen** [microsoft/TypeScript#44331]
# Bug Report
The compiler should throw a "symbol not found" error when a symbol can't be found in the current file. However, when a symbol exists in another file, but is not exported, the error doesn't show up, compilation succeeds, and a runtime error is encountered instead.

### 🔎 Search Terms
tsc doesn't throw build errors

### 🕗 Version & Regression Information

From TSC 4.2, now on 4.3.2 it's still here. On 4.4.0-dev.20210529 it's still here

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about function discovery

### ⏯ Playground Link:

```ts repro
 // @filename: fileb.ts
function dostuff(){
    console.log("hello, world")
}

// @filename: filea.ts
class test {
    constructor(){
        dostuff()
    }
}
```

[Workbench Repro](https://www.typescriptlang.org/dev/bug-workbench/?ssl=11&ssc=2&pln=1&pc=1#code/AQehwAQMwSwGwKYDsCGBbBAuYtECMA6AFwGcAoKAVyQGMiYB7JYAEwZKMqigAoBKAN5lgI4DSYkGiAnAYBzHgCIAFgjiyANMADuDAE5wWivmQC+ZMmEi5k6LDngIUxcjTgoSJYEQQdgQ0TEJIj1KOn1+AMDRNg4uXhNA81MgA)

### 💻 Code

See code above. This compiles without errors. When the function `dostuff()` is exported from `fileb.ts`, only then is there an error when I try to compile filea.ts

### 🙁 Actual behavior

filea.ts compiles without throwing an error. This is weird, because filea.ts has no import for dostuff()

```ts
 // @filename: fileb.ts
export function dostuff(){
    console.log("hello, world")
}

// @filename: filea.ts
class test {
    constructor(){
        dostuff()
    }
}
```
This does error like it should.

### 🙂 Expected behavior

TSC should throw a "symbol not found" error on compilation.

**Comment by whzx5byb** [comment]
This is an intended behavior. Your `filea.ts` and `fileb.ts` are treated as parts of the global namespace instead of modules. 
See https://www.typescriptlang.org/docs/handbook/2/modules.html

**Comment by tanevanwifferen** [comment]
Ah ok I get it. Thanks :)
